crates/transaction-pool/src/pool/blob.rs

```diff
@@ -101,6 +100,51 @@ impl<T: PoolTransaction> BlobTransactions<T> {
         self.by_id.len()
     }
 
+    /// Returns whether the pool is empty
+    #[cfg(test)]
+    #[allow(unused)]
+    pub(crate) fn is_empty(&self) -> bool {
+        self.by_id.is_empty()
+    }
+
+    /// Returns all transactions that satisfy the given blob fee.
+    fn satisfy_blob_fee_ids(&self, blob_fee: u128) -> Vec<TransactionId> {
+        let mut transactions = Vec::new();
+        {
+            let mut iter = self.by_id.iter().peekable();
+
+            while let Some((id, tx)) = iter.next() {
+                if tx.transaction.max_fee_per_blob_gas() < Some(blob_fee) {
+                    // still parked in blob pool -> skip descendant transactions
+                    'this: while let Some((peek, _)) = iter.peek() {
+                        if peek.sender != id.sender {
+                            break 'this
+                        }
+                        iter.next();
+                    }
+                } else {
+                    transactions.push(*id);
+                }
+            }
+        }
+        transactions
+    }
+
+    /// Removes all transactions and their dependent transaction from the subpool that no longer
+    /// satisfy the given blobfee.
+    ///
+    /// Note: the transactions are not returned in a particular order.
+    pub(crate) fn enforce_blob_fee(&mut self, blob_fee: u128) -> Vec<Arc<ValidPoolTransaction<T>>> {
+        let to_remove = self.satisfy_blob_fee_ids(blob_fee);
+
+        let mut removed = Vec::with_capacity(to_remove.len());
+        for id in to_remove {
+            removed.push(self.remove_transaction(&id).expect("transaction exists"));
+        }
+
+        removed
+    }
```

Comment:
~~Isn't this step entirely redundant?~~

~~first, we remove them here, only to then add them back into this pool?~~

oh I see why we need this, because this is called when the blob fee decreased. however the docs are wrong here, because this is supposed to return txs that satisfy the blob fee.

we also need to combine this here with the basefee, because otherwise we might end up moving the transaction twice, so this should take both the basefee and the blobfee as argument and return all transactions that satisfy both.

Reply:
I added `PendingFees` which contains the base fee and blob fee, now this is passed into a method `enforce_pending_fees`. The tx should only move now if the tx satisfies both the basefee and blobfee. also added a test for it